File: tutorials/testing/java/tut_worldmap_p3.java
import java.util.ArrayList;
import java.util.Map;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.Country;
import bridges.base.WorldMap;
import bridges.base.Color;
import bridges.base.GraphAdjListSimple;


// This tutorial illustrates how to access the data of the World map with 
// boundaries; the countries are filled with default color
public class tut_worldmap_p3 {
	public static void main(String[] args) throws Exception {

		// create Bridges object
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", 
												"YOUR_API_KEY");
		// set title, description
		bridges.setTitle("Tutorial: Demonstrating overlaying World Map with Cities at their locations");
		bridges.setDescription("Uses graph vertices and lat/long coords to locate the cities");

		// get data source object
		DataSource ds = bridges.getDataSource();

		GraphAdjListSimple<String> gr = new GraphAdjListSimple<String>();

		gr.addVertex("Charlotte", "Charlotte,NC, USA");
			gr.getVertex("Charlotte").setLocation(-80.8431, 35.2271);
			gr.getVertex("Charlotte").setColor(new Color("crimson"));
			gr.getVertex("Charlotte").setSize(2.);
			gr.getVertex("Charlotte").setLabel("Charlotte, NC");
		gr.addVertex("Paris", "Paris, France");
			gr.getVertex("Paris").setLocation(2.3514, 48.8576);
			gr.getVertex("Paris").setColor(new Color("crimson"));
			gr.getVertex("Paris").setSize(2.);
		gr.addVertex("Berlin", "Berlin, Germany");
			gr.getVertex("Berlin").setLocation(13.4, 52.52);
			gr.getVertex("Berlin").setColor(new Color("crimson"));
			gr.getVertex("Berlin").setSize(2.);
		gr.addVertex("New Delhi", "New Delhi, India");
			gr.getVertex("New Delhi").setLocation(77.208, 28.614);
			gr.getVertex("New Delhi").setColor(new Color("crimson"));
			gr.getVertex("New Delhi").setSize(2.);
		gr.addVertex("Buenos Aires", "Buenos Aires, Argenina");
			gr.getVertex("Buenos Aires").setLocation(-58.382, -34.60);
			gr.getVertex("Buenos Aires").setColor(new Color("crimson"));
			gr.getVertex("Buenos Aires").setSize(2.);
		gr.addVertex("Esperanza Station, Antarctica", "Esperanza Station, Antarctica");
			gr.getVertex("Esperanza Station, Antarctica").setLocation(-56.59, -63.24);
			gr.getVertex("Esperanza Station, Antarctica").setColor(new Color("crimson"));
			gr.getVertex("Esperanza Station, Antarctica").setSize(2.);


		WorldMap wm = new WorldMap();
		bridges.setMap(wm);
		bridges.setLabelFlag(true);

		// set data structure
		bridges.setDataStructure(gr);

		// visualize
		bridges.visualize();
	}
}

File: tutorials/testing/python/tut_worldmap_p3.py
from bridges.bridges import *
from bridges.data_src_dependent import *
from bridges.dl_element import *
from bridges.world_map import *
from bridges.graph_adj_list import *

def main():
    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    # set title, description
    bridges.set_title("Overlay cities on a World Map ")
    bridges.set_description("Draws all countries and place some cities on the map")

    # with no arguments to the function, gets all countries of the world
    country_data = get_world_map_data()
    wm = WorldMap()

	# lets add  a points to the map to illustrate how to identify cities 
    # we will use graph vertices for this

    gr = GraphAdjList()
    gr.add_vertex("Charlotte", "Charlotte,NC")
    gr.get_vertex("Charlotte").set_location(-80.8431, 35.2271)
    gr.get_vertex("Charlotte").color = "crimson"
    gr.get_vertex("Charlotte").size = 2.

    gr.add_vertex("Paris", "Paris, France")
    gr.get_vertex("Paris").set_location(2.3514, 48.8576)
    gr.get_vertex("Paris").color = "crimson"
    gr.get_vertex("Paris").size = 2.

    gr.add_vertex("Berlin", "Berlin, Germany")
    gr.get_vertex("Berlin").set_location(13.4, 52.52)
    gr.get_vertex("Berlin").color = "crimson"
    gr.get_vertex("Berlin").size = 2.

    gr.add_vertex("New Delhi", "New Delhi, India")
    gr.get_vertex("New Delhi").set_location(77.208, 28.614)
    gr.get_vertex("New Delhi").color = "crimson"
    gr.get_vertex("New Delhi").size = 2.

    gr.add_vertex("Buenos Aires", "Buenos Aires, Argenina")
    gr.get_vertex("Buenos Aires").set_location(-58.382, -34.60)
    gr.get_vertex("Buenos Aires").color = "crimson"
    gr.get_vertex("Buenos Aires").size = 2.

    gr.add_vertex("Esperanza Station, Antarctica", "Esperanza Station, Antarctica")
    gr.get_vertex("Esperanza Station, Antarctica").set_location(-56.59, -63.24)
    gr.get_vertex("Esperanza Station, Antarctica").color = "crimson"
    gr.get_vertex("Esperanza Station, Antarctica").size = 2.

	# add the map overlay
    bridges.set_map(wm)

    #visualizes world map
    bridges.set_data_structure(gr)
    bridges.visualize()

if __name__ == '__main__':
    main()
